### PR TITLE
style: ensure all error messages start with lowercase

### PR DESCRIPTION
**Pull request by valeriiashapoval**
## Description
(https://issues.redhat.com/browse/MGDSTRM-9942)

## Checklist (Definition of Done)

- [X] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [X] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

**Review by machi1990 — COMMENTED** [review]
@miguelsorianod I remember that you had some possible comments on automating the detection part of this here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1344#discussion_r997825493 can you have a look at this PR and see how your suggestions could be incorporated onto this P?

**Review by machi1990 — CHANGES_REQUESTED** [review]
CI is failing. Can you have a look at https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/actions/runs/3322803896/jobs/5492340789 and follow on the suggested approach?

**Comment by ziccardi** [comment]
> @miguelsorianod I remember that you had some possible comments on automating the detection part of this here [#1344 (comment)](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1344#discussion_r997825493) can you have a look at this PR and see how your suggestions could be incorporated onto this P?

I believe the easiest way is using `staticcheck` together with golangci-lint: [staticcheck](https://staticcheck.io/docs/checks/#ST1005)

**Comment by lburgazzoli** [comment]
it looks like some of the changes are related to code that is generated by the open api generator, isn't it ?

**Comment by miguelsorianod** [comment]
> > @miguelsorianod I remember that you had some possible comments on automating the detection part of this here [#1344 (comment)](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1344#discussion_r997825493) can you have a look at this PR and see how your suggestions could be incorporated onto this P?
> 
> I believe the easiest way is using `staticcheck` together with golangci-lint: [staticcheck](https://staticcheck.io/docs/checks/#ST1005)

That is another linter where it can be configured yes. As I mentioned in my comment I understand it will only detect cases of fmt.Errorf so it won't cover everything, but it is an improvement so let's enable it.

**Comment by miguelsorianod** [comment]
> it looks like some of the changes are related to code that is generated by the open api generator, isn't it ?

We shouldn't be applying changes on the autogenerated code.

**Comment by valeriiashapoval** [comment]
> CI is failing. Can you have a look at https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/actions/runs/3322803896/jobs/5492340789 and follow on the suggested approach?

thanks @machi1990 ! CI is passing now, I also added staticcheck to the linter configuration as @miguelsorianod has suggested. Thank you!

**Review by machi1990 — COMMENTED** [review]
The code change lgtm, thanks @valeriiashapoval 

I do not think the API consumer are relying on the `error.reason`. At the moment, our main consumers are mainly DevExp team, this might be something we've to confirm with them before we do merge this.

Also lets get a review from connector @dhirajsb @lburgazzoli

**Comment by valeriiashapoval** [comment]
thanks @machi1990 ! should be rebased now.

File: .golangci.yml
Comment (miguelsorianod) [inline .golangci.yml:16]: This doesn't seem to be doing anything at all.

Even when adding this line running `make lint` locally it does not detect any new errors.

Steps to reproduce:
1. Checkout `main` branch
1. Run `make lint`. Observe how no errors are detected
1. Add the `staticcheck` line as done in this PR
1. Run `make lint`. Observe how no errors are detected

If that line really did something running the previous steps would detect errors, as in the `main` branch the changes of this PR are not available.

File: internal/connector/internal/api/dbapi/connector_type.go
Comment (miguelsorianod) [inline internal/connector/internal/api/dbapi/connector_type.go:128]: How have this kind of needed changes been detected?

I understand that by checking manually and not by running any linter, right?

File: .golangci.yml
Comment (ziccardi) [reply .golangci.yml:16]: I believe the reason is this: https://github.com/golangci/golangci-lint/discussions/1989

Instead of `staticcheck`, we should enable `stylecheck`.
Then, this configuration limits stylecheck to check only error capitalization:
```
  stylecheck:
    checks: ["ST1005"]
```
(see https://golangci-lint.run/usage/linters/#stylecheck)

Comment (valeriiashapoval) [reply .golangci.yml]: Thanks @miguelsorianod @ziccardi ! Just pushed the suggested changes.

File: internal/connector/internal/cmd/vault/list.go
Comment (ziccardi) [inline internal/connector/internal/cmd/vault/list.go:23]: The rule applies to error messages. The reason is that error messages are usually appended/included in the log statement, and uppercases or punctuation in error messages would make to log statement weird.

Log statements, on the other hand, are self-contained and don't have that restriction.

File: internal/connector/test/integration/cleanup/main.go
Comment (ziccardi) [inline internal/connector/test/integration/cleanup/main.go:48]: This does not needs to be changed (see previous comment)

File: internal/connector/test/integration/cleanup/main.go
Comment (ziccardi) [inline internal/connector/test/integration/cleanup/main.go:76]: This does not needs to be changed (see previous comment)

File: internal/connector/test/integration/cucumber_steps.go
Comment (ziccardi) [inline internal/connector/test/integration/cucumber_steps.go:118]: This does not needs to be changed (see previous comment)

**Review by ziccardi — CHANGES_REQUESTED** [review]
Several log statements (I just flagged some) have been lowercase.
That's not needed: the rule applies only to error messages: since error messages are usually embedded in a log statement, starting it with uppercase would make the log statement look weird.

File: internal/kafka/internal/handlers/kafka_instance_types_test.go
Comment (miguelsorianod) [inline internal/kafka/internal/handlers/kafka_instance_types_test.go:72]: This should be lowercase

File: internal/connector/test/integration/cucumber_steps.go
Comment (valeriiashapoval) [reply internal/connector/test/integration/cucumber_steps.go]: thanks @ziccardi . Should be addressed now.

File: internal/kafka/internal/handlers/kafka_instance_types_test.go
Comment (valeriiashapoval) [reply internal/kafka/internal/handlers/kafka_instance_types_test.go]: thanks @miguelsorianod ! Just added the change.

File: internal/connector/test/integration/cucumber_steps.go
Comment (ziccardi) [reply internal/connector/test/integration/cucumber_steps.go]: @valeriiashapoval There are still several logs that have been lowercased.

Comment (valeriiashapoval) [reply internal/connector/test/integration/cucumber_steps.go]: thanks @ziccardi . all logs should be fixed now.

File: internal/kafka/internal/services/quota/quota_management_list_service_test.go
Comment (ziccardi) [inline internal/kafka/internal/services/quota/quota_management_list_service_test.go:358]:
```suggestion
			wantErr: errors.InsufficientQuotaError("insufficient quota"),
```

Comment (valeriiashapoval) [reply internal/kafka/internal/services/quota/quota_management_list_service_test.go]: thanks @ziccardi ! just changed it.

File: internal/connector/internal/api/dbapi/connector_type.go
Comment (ziccardi) [reply internal/connector/internal/api/dbapi/connector_type.go:128]: @valeriiashapoval As Miguel asked, did we have this manually or automatically detected?

**Comment by valeriiashapoval** [comment]
> Changes LGTM. I will wait for other reviewers to approve too.
> 
> Thanks @valeriiashapoval !

Thanks @ziccardi !

**Comment by miguelsorianod** [comment]
looks good to me. Please squash the issues and rebase the PR

File: internal/connector/internal/api/dbapi/connector_type.go
Comment (miguelsorianod) [reply internal/connector/internal/api/dbapi/connector_type.go:128]: hi @valeriiashapoval please remember to answer to this. Thanks

**Comment by valeriiashapoval** [comment]
> looks good to me. Please squash the issues and rebase the PR

Thanks @miguelsorianod . Just rebased and squashed the commits, thanks!

**Review by miguelsorianod — APPROVED** [review]
Looks good to me. Let's wait until the tests pass and then merge

**Comment by miguelsorianod** [comment]
It needs a rebase again

**Comment by miguelsorianod** [comment]
The tests are also failing due to unmatching strings in some of them

**Comment by valeriiashapoval** [comment]
thanks @miguelsorianod! just addressed the issue with the tests and rebased again.

**Review by miguelsorianod — APPROVED** [review]

**Pull request merged**
